nx/source/services/clkrst.c

```diff
@@ -0,0 +1,152 @@
+#include "types.h"
+#include "result.h"
+#include "arm/atomics.h"
+#include "kernel/ipc.h"
+#include "services/pcv.h"
+#include "runtime/hosversion.h"
+#include "services/sm.h"
+#include "services/clkrst.h"
+
+static Service g_clkrstSrv;
+static u64 g_refCnt;
+
+Result clkrstInitialize(void) {
+    if(hosversionBefore(8,0,0)) {
+        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
+    }
+
+    atomicIncrement64(&g_refCnt);
+
+    if (serviceIsActive(&g_clkrstSrv)) {
+        return 0;
+    }
+
+    Result rc = smGetService(&g_clkrstSrv, "clkrst");
+
+    if (R_FAILED(rc)) {
+        clkrstExit();
+    }
+
+    return rc;
+}
+
+
+void clkrstExit(void) {
+    if (atomicDecrement64(&g_refCnt) == 0) {
+        serviceClose(&g_clkrstSrv);
+    }
+}
+
+Result clkrstOpenSession(ClkrstSession* session_out, PcvModule module) {
+    IpcCommand c;
+    ipcInitialize(&c);
+
+    struct {
+        u64 magic;
+        u64 cmd_id;
+        u32 module;
+        u32 unk;
+    } *raw;
+
+    raw = ipcPrepareHeader(&c, sizeof(*raw));
+
+    raw->magic = SFCI_MAGIC;
+    raw->cmd_id = 0;
+    raw->module = 0x40000001 + module;
```

Comment:
Right, while they all map for the ones currently defined in pcv.h, it'd be breaking change in the future if we were to add all of them. @yellows8 Would you like all the modules defined ? It would be an enum that would be common to clkrst and rgltr API

Reply:
sure

Reply:
Should i merge clkrst funcs in pcv.h/.c then, or what's the naming scheme you'd suggest?

Reply:
Just add the enum in pcv.h.